Ngoc_Bao/myweb/app/Http/Controllers/Admin/ProductController.php

```diff
@@ -3,84 +3,215 @@
 namespace App\Http\Controllers\Admin;
 
 use App\Http\Controllers\Controller;
+use App\Http\Controllers\Auth;
 use Illuminate\Http\Request;
 use App\Models\Product;
 use App\Models\Category;
+use App\Models\ProductHistory;
+use App\Models\ProductHistoryDetail;
 
 class ProductController extends Controller
 {
-    protected $productmodel;
-    protected $categorymodel;
+    protected $modelProduct;
+    protected $modelCategory;
+    protected $modelProductHistory;
+    protected $modelProductHistoryDetail;
 
-    public function __construct(Product $product, Category $category)
+    public function __construct(Product $product, Category $category, ProductHistory $producthistory, ProductHistoryDetail $producthistorydetail)
     {
-        $this->productmodel = $product;
-        $this->categorymodel = $category;
+        $this->modelProduct = $product;
+        $this->modelCategory = $category;
+        $this->modelProductHistory = $producthistory;
+        $this->modelProductHistoryDetail = $producthistorydetail;
     }
 
     public function index()
     {
         //return view get data
-        $products = $this->productmodel->paginate(10);
-        return view('admin.products.index',['products'=>$products]);
+        $products = $this->modelProduct->paginate(10);
+        $user = \Auth::user();
```

Comment:
```suggestion
        $user = auth()->user();
```
có thể dùng helper này nhé e